.github/workflows/performance-tests.yml

```diff
@@ -30,19 +30,33 @@ jobs:
     steps:
       - name: Checkout repository
         uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
 
       - name: Setup Python
         uses: actions/setup-python@v4
         with:
           python-version: 3.10.13
           cache: pip
 
-      - name: Install pip dependencies
+      # TODO(danh): switch to actually checking CircleCI builds
+      - name: Get version for latest master image
+        id: master-version
+        run: |
+          git checkout main
+          git rev-parse HEAD > .master-image-version
+          echo "Looking for master image version $(cat .master-image-version)"
+          docker pull "determinedai/determined-master:$(cat .master-image-version)"
+          echo "text=$(cat .master-image-version)" >> "$GITHUB_OUTPUT"
```

Comment:
If you're only ever going to use this value in steps within the same job, I'd probably make it easier to use by putting it as a variable in GITHUB_ENV instead of setting it as step output. :)  Up to you, though.